Services/WebServices/SOAP/classes/class.ilAbstractSoapMethod.php

```diff
@@ -0,0 +1,80 @@
+<?php
+require_once(__DIR__ . '/ilSoapMethod.php');
+require_once(__DIR__ . '/class.ilSoapPluginException.php');
+require_once('./webservice/soap/classes/class.ilSoapAdministration.php');
+
+/**
+ * Class ilAbstractSoapMethod
+ *
+ * Base class for soap methods of SoapHook plugins.
+ * Throw a ilSoapPluginException in your business logic in case of errors. The plugin hook catches these exceptions
+ * and returns the exception messages to the SOAP caller.
+ *
+ * @author Stefan Wanzenried <sw@studer-raimann.ch>
+ */
+abstract class ilAbstractSoapMethod extends ilSoapAdministration implements ilSoapMethod {
+
+	public function __construct() {
+		parent::__construct(true);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getServiceStyle() {
+		return 'rpc';
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getServiceUse() {
+		return 'encoded';
+	}
+
+	/**
+	 * Use this method at the beginning of your execute() method to check if the provided session ID is valid.
+	 * This method wraps around ilSoapAdministration::initAuth() and ilSoapAdministration::initILIAS()
+	 * which are both required in order to handle the request.
+	 *
+	 * @param string $session_id
+	 * @throws ilSoapPluginException
+	 */
+	protected function initIliasAndCheckSession($session_id) {
+		$this->initAuth($session_id);
+		$this->initIlias();
+		if (!$this->__checkSession($session_id)) {
+			throw new ilSoapPluginException('Invalid session-ID provided');
```

Comment:
throw new ilSoapPluginException($this->__getMessage()); will keep the original error message from __checkSession